druid/src/widget/mod.rs

```diff
@@ -252,6 +252,11 @@ pub trait Widget<T> {
     fn id(&self) -> Option<WidgetId> {
         None
     }
+
+    /// Get the (verbose) type name of the widget for debugging purposes.
+    fn type_name(&self) -> &'static str {
```

Comment:
on the `Box` impl I would call this method on the inner type; otherwise all boxed widgets will be `Box<dyn Widget<_>>`. Although maybe that's fine?

Reply:
oh one other request: can we do `#[doc(hidden)]` here, and can we say "you should not override this method"?

We'll come up with a better way to do this eventually.